### DISCUSSION TITLE
Definition of Done

### DISCUSSION THREAD

**Discussion by sytskevanhasselt** [Klantinteractie-Servicesysteem/KISS-frontend#36] [General]
De Definition of Done is vastgesteld, en overgenomen [in de documentatie](https://github.com/Klantinteractie-Servicesysteem/.github/blob/main/docs/DEFINITIONOFDONE.md).
 

De DoD is een waarborg voor (een deel van) de [architectuurkaders](https://icattnl.sharepoint.com/:x:/r/sites/EXTERN-Conduction/Gedeelde%20documenten/KISS/standaarden%20en%20kwaliteitsnormen%20(DOD)/Bijlage%206.11.2%20Selectietool%20architectuurkaders%20-%20KISS.v0.1.xlsx?d=w34edd875af554db3a05ba76bac1ca101&csf=1&web=1&e=UdfMm1)

**Comment by rubenvdlinde** [comment]
Conduction en ICAT doen voorstel op 23 mei. 

Wie hebben we nodig?
- [ ] Stuurgroep  of Project leider
- [ ] Dat iedereen van de kerngroep online reageert vóór 30 mei.
- [ ] 2 juni vaststellen in ontwerpgroep

**Comment by frankmijn** [comment]
Het criterium 'UX (user experience) voldoet aan NL design system, gereviewd en geaccepteerd.' lijkt lastig in te vullen als ik kijk naar https://github.com/Klantinteractie-Servicesysteem/KISS-frontend/discussions/39. 

Er staat wel 'Accepatie creteria zijn afgevink door PO' maar wat ik mis, is een algehele goedkeuring van de PO dat de functionaliteit invulling geeft aan de user story. Kan ook dat dit hiermee bedoeld wordt.

Ik mis een criterium dat het resultaat van de user story op een acceptatie-omgeving beschikbaar is.

Is binnen het team alle expertise aanwezig om deze criteria af te vinken?

**Reply by PascalIcatt** [reply]
Algehele goedkeuring van de PO op functionaliteit zal als eerste gebeuren bij de developer op localhost.
Hierbij worden alle acceptatie- en functionele DoD criteria (bv WCAG score check) gecontroleerd en afgevinkt. (templating hiervoor dient aan de stories toegevoegd te worden)

Hierna wordt een codereview gedaan via een Pull Request, hierin wordt al een Build gedraaid die de automatische tests bij elke commit controleert.

Zodra het Pull Request door een 2e developer is goedgekeurd (nadat opmerkingen/bugs/testresultaten zijn verwerkt) wordt de hele omgeving gedeployed naar Acceptatie.
Op acceptatie zullen Pentests plaatsvinden na deployment.

Zodra ook deze afgerond zijn, is voldaan aan alle criteria en is de story volgens de definitie 'done'.
Eventuele aanpassingen obv de pentest resultaten worden in de volgende Release meegenomen.

Binnen het team zijn alle expertises en kennis aanwezig.

**Reply by PascalIcatt** [reply]
Het NL design criterium is inderdaad niet zo makkelijk controleer en invulbaar.
Dit is echter wel opgenomen als wens in de selectietool architectuurkaders, dit zal terug komen in de ontwerpsessies en op basis van best-effort opgenomen worden in de DoD

**Comment by sytskevanhasselt** [comment]
Deze discussie is gesloten. Het resultaat is opgenomen [in contributing.md in de repository](https://github.com/Klantinteractie-Servicesysteem/KISS-frontend/blob/acceptance/CONTRIBUTING.md) (link gaat naar de acceptance branche)
